### PR TITLE
python3Packages.regex: 2020.11.13 -> 2021.4.4

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 2021.4.4

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118522` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>home-assistant</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
@fabaff can you please manually rebase it on staging? the version there is a bit newer.

**Comment by fabaff** [comment]
@ofborg eval

File: pkgs/development/python-modules/regex/default.nix [hunk 1]
```diff
@@ -4,14 +4,21 @@
...
   src = fetchPypi {
     inherit pname version;
     sha256 = "4b8a1fb724904139149a43e172850f35aa6ea97fb0545244dc0b805e0154ed68";
+=======
+  version = "2021.4.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-Uro9P5uULEnX5LwQW7KFUcRAZfE5plBiq3kSvvEMmvs=";
+>>>>>>> a8abf713a3b (python3Packages.regex: 2020.11.13 -> 2021.4.4)
```
Comment (mweinelt) [inline pkgs/development/python-modules/regex/default.nix:21]: This looks like a rebase gone wrong?

Comment (fabaff) [reply pkgs/development/python-modules/regex/default.nix]: Yes, it did. Thanks.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118522` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>home-assistant</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/regex/default.nix [hunk 2]
```diff
@@ -22,10 +21,12 @@ buildPythonPackage rec {
   # No tests in archive
   doCheck = false;
```
Comment (mweinelt) [inline pkgs/development/python-modules/regex/default.nix:22]: Can we fetch from bitbucket and get the tests running?

Comment (fabaff) [reply pkgs/development/python-modules/regex/default.nix]: Test are enabled now.

File: pkgs/development/python-modules/regex/default.nix [hunk 3]
```diff
@@ -2,30 +2,30 @@
...
   };
 
-  postCheck = ''
-    echo "We now run tests ourselves, since the setuptools installer doesn't."
-    ${python.interpreter} -c 'import test_regex; test_regex.test_main();'
+  # Sources for different Python releases are located in same folder
+  checkPhase = ''
+    rm -r ${if !isPy27 then "regex_2" else "regex_3"}
+    ${python.interpreter} -m unittest
   '';
```
Comment (mweinelt) [inline pkgs/development/python-modules/regex/default.nix:21]:
```
running install tests

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
```

For `python2Packages.regex` it doesn't seem to pick up the tests.

**Comment by mweinelt** [comment]
Merged into my local working tree for python-unstable. Thank you.

**Pull request closed without merging**